src/ucp/wireup/wireup.c

```diff
@@ -419,9 +419,12 @@ ucp_wireup_connect_local(ucp_ep_h ep,
 
 void ucp_wireup_remote_connect_lanes(ucp_ep_h ep, int ready)
 {
-    ucp_lane_index_t lane;
+    ucp_lane_index_t i, lane;
 
-    for (lane = 0; lane < ucp_ep_num_lanes(ep); ++lane) {
+    for (i = 0; i < ucp_ep_num_lanes(ep);) {
+        /* Connect lanes in reverse ordering to avoid replaying requests with
+         * multi fragment protocol from one to other wireup lane. */
+        lane = ucp_ep_num_lanes(ep) - (++i);
```

Comment:
1. why reverse order helps here?
2.  IMO this looks better:
```
int lane;
for (lane = ucp_ep_num_lanes(ep) - 1; lane >= 0; --lane) {
```

Reply:
> why reverse order helps here?

no moving request from one wireup ep to another. If ordering of connecting lanes is the same as `request->send.lane` then the request will go via all wireup endpoints and their pending queues

Reply:
> IMO this looks better

My loop is more positive :) and no type cast.. but need to deside ordering approach first